app/src/components/Map/AreasPanel.jsx

```diff
@@ -38,10 +43,15 @@ class AreasPanel extends Component {
 
   onNameChange(event) {
     const name = event.target.value;
-    this.setState({ name });
+    this.props.saveEditingArea({ name });
+  }
+
+  onColorChange(color) {
+    this.props.saveEditingArea({ color });
   }
 
   render() {
+    const { name, color } = this.props.editingArea;
     return (
       (this.props.drawing ?
```

Comment:
Rule of thumb: only use the trenary operator if each part of it takes 1 line at most. 

When you have more than 1 line, the trenary operator is easy to miss. In those situations, use a traditional if/else clause.